packages/terra-data-grid/src/FlowsheetDataGrid.jsx

```diff
@@ -229,33 +231,56 @@ function FlowsheetDataGrid(props) {
     }
   }, [intl, rows, columns, setCellSelectionAriaLiveMessage]);
 
-  const selectCellRange = useCallback((rowIndex, columnIndex) => {
+  const selectCellRange = useCallback((rowId, columnId, sectionId) => {
     if (anchorCell.current === null) {
       return;
     }
 
-    const anchorRowIndex = rows.findIndex(row => row.id === anchorCell.current.rowId);
+    const rowsToSelect = flowsheetSections?.find(section => section.id === sectionId)?.rows ?? rows;
+    const anchorRowIndex = rowsToSelect.findIndex(row => row.id === anchorCell.current.rowId);
     const anchorColumnIndex = columns.findIndex(col => col.id === anchorCell.current.columnId);
 
+    const rowIndex = rowsToSelect.findIndex(row => row.id === rowId);
+    const columnIndex = columns.findIndex(col => col.id === columnId);
+
     // Determine the boundaries of selected region.
-    const rowIndexTopBound = Math.min(anchorRowIndex, rowIndex - 1);
-    const rowIndexBottomBound = Math.max(anchorRowIndex, rowIndex - 1);
+    let rowIndexTopBound = Math.min(anchorRowIndex, rowIndex);
+    let rowIndexBottomBound = Math.max(anchorRowIndex, rowIndex);
     const columnIndexLeftBound = Math.min(anchorColumnIndex, columnIndex);
     const columnIndexRightBound = Math.max(anchorColumnIndex, columnIndex);
 
+    if (flowsheetSections) {
+      const anchorSection = flowsheetSections.find(section => section.id === anchorCell.current.sectionId);
+      const selectedSection = flowsheetSections.find(section => section.id === sectionId);
+
+      if (anchorSection.id !== selectedSection.id) {
```

Comment:
The sectionId is being passed as an argument to the method.  Why do we need to use the selectedSection object here?